src/sbml/conversion/SBMLLevelVersionConverter.cpp

```diff
@@ -1321,6 +1313,32 @@ SBMLLevelVersionConverter::collectSpeciesReferenceIds()
 
   return srids;
 }
+
+
+void
+SBMLLevelVersionConverter::updateUnitMap(SBMLDocument* d)
+{
+  if (d->getModel()->isPopulatedListFormulaUnitsData())
```

Comment:
it would be good to check whether a model is set (should be the case, but since this is a function we might want to check)